SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/AreaManagerNamingViewModel.cs

```diff
@@ -0,0 +1,136 @@
+                return;
+            }
+
+            var area = GetArea(Player);
+
+            if (AreaManagerViewModel.GetIsTemplateArea(area))
+            {
+                SendMessageToPC(Player, $"Please do not create a template area of a template area. Go to the area that is part of the core module and template that instead.");
+                Gui.TogglePlayerWindow(Player, GuiWindowType.AreaManagerNaming);
+                return;
+            }
+
+            // Stores transition targets to restore after CreateArea is called.
+            for (var x = GetFirstObjectInArea(area); GetIsObjectValid(x); x = GetNextObjectInArea(area))
+            {
+                if (GetTransitionTarget(x) != OBJECT_INVALID)
+                    SetLocalObject(x, "ORIGINAL_TRANSITION_TARGET", GetTransitionTarget(x));
+            }
+
+            // Generating a random number for the area tag allows for duplicates of the same area without messing up the way in which objects are saved
+            // in each template area. It's very improbable that two duplicate areas would have the exact same tag name and randomize the exact same number.
+            // However, it's less improbable that two areas may have the exact same tag names. For example, if a DM names all his template areas "Ziya's Corellian Event Zone 1" then
+            // there's a high chance all his areas would be "tmp_ziyascorelli" since it caps at 16 characters, which could mess up the ways objects are saved
+            // to each template area. By adding a random 3-digit number to each template area's tag, it makes that duplication highly improbable.
+            var randomNumber = Random(999);
+            var targetArea = CreateArea(GetResRef(area), "tmp_" + randomNumber + GetTag(area), templateAreaName);
+            SetLocalBool(targetArea, "IS_TEMPLATE_AREA", true);
+
+            // This creates the area and stores it in the database so that it can be loaded with each module load.
+            var templateAreaTag = GetTag(targetArea);
+            var templateAreaResRef = GetResRef(area);
+
+            var templateArea = new TemplateArea
+            {
+                TemplateAreaName = templateAreaName,
+                TemplateAreaTag = templateAreaTag,
+                TemplateAreaResRef = templateAreaResRef,
+            };
+            DB.Set(templateArea);
+
+            // This deletes all of the creatures, waypoints, and transitions. Since many of our transitions are
+            // placeables with the DESTINATION tag, the GetObjectType(x) == ObjectType.Placeable && GetLocalBool(x, "DESTINATION")) line
+            // deletes placeables that serve as transitions, which would solve bugs like the placeable teleport
+            // in Jim's Cantina surviving to the template area and teleporting people to weird places. 
+            for (var x = GetFirstObjectInArea(targetArea); GetIsObjectValid(x); x = GetNextObjectInArea(targetArea))
```

Comment:
You're doing this logic in multiple places. Add this method to the service and call it to reduce code duplication.